sdk/cosmos/azure-cosmos/azure/cosmos/aio/_cosmos_client.py

```diff
@@ -224,7 +225,18 @@ async def create_database(  # pylint: disable=redefined-builtin
         response_hook = kwargs.pop('response_hook', None)
         offer_throughput = kwargs.pop('offer_throughput', None)
         if offer_throughput is not None:
-            request_options["offerThroughput"] = offer_throughput
+            try:
+                if offer_throughput.auto_scale_max_throughput:
+                    request_options['autoUpgradePolicy'] = _stringify_auto_scale(offer=offer_throughput)
+                elif offer_throughput.auto_scale_increment_percent:
+                    raise ValueError("auto_scale_max_throughput must be supplied in "
+                                     "conjunction with auto_scale_increment_percent")
+                elif offer_throughput.offer_throughput:
+                    request_options["offerThroughput"] = offer_throughput.offer_throughput
+
+            except AttributeError:
+                if isinstance(offer_throughput, int):
+                    request_options["offerThroughput"] = offer_throughput
```

Comment:
This accepts either an integer or complex object?  Why are we doing that? And why is `offer_throughput.offer_throughput` an acceptable input?

Reply:
@tjprescott the method traditionally only accepted an int but since we want to use the same option for autoscale properties, we had to change the signature to accept the object as well

This decision was made since otherwise we would be creating a new object with only the two autoscale properties and adding additional options to the method that are throughput related - this way we instead use what we already had while still adding the new functionality